### DISCUSSION TITLE
Starcoin 2.0 with FlexiDag pre-mining user guide

### DISCUSSION THREAD

**Discussion by sanlee42** [starcoinorg/starcoin#3983] [General]
# Starcoin DAG-Net Mining Guide

### Prerequisites
- **Platform**: PC with Windows, macOS, or Linux.
- **Operating System**: Ensure you have a compatible version (e.g., Windows 10, macOS Catalina, Ubuntu 20.04).
- **Required Software**:
  - Git: For source code management.
  - Rust and Cargo: For compiling the source code.

### Steps
1. **Open Terminal**:
   - Windows: Open Command Prompt or PowerShell.
   - macOS/Linux: Open the Terminal application.

2. **Clone the Repository**:
   ```bash
   git clone https://github.com/starcoinorg/starcoin.git
   ```
3. **Switch to the Desired Branch**:
    ```
    git checkout origin/dag-test-net:
    ```
4. **Build from source** :
    ```
    cargo build --release
    ```

## Download from Release

**Direct Download**:
- Visit the [Starcoin Release Page](https://github.com/starcoinorg/starcoin/releases/tag/dag-test-net-v2.0.1).
- Choose the appropriate version for your operating system.
- Download and extract the release files to a known directory.

## Start Node & Join DAG Network for Mining

```bash
# Setting the miner client thread number to 4
./target/debug/starcoin -n dag-net:24--genesis-config  halley --miner-thread 4 console
```
## Check chain status
``` bash
starcoin% chain info
```

**Comment by sanlee42** [comment]
###  Now we can join dag-net for mining in public p2p network
* Connect to the seed 
``` 
./starcoin -n dag-net:24 --genesis-config  halley   --seed /ip4/139.59.110.185/tcp/9840/p2p/12D3KooWEXmxsqvjaQj5ChB8ckaRAyYfGBjSHoo3byBV1BjpJGbK  console

```
* Check node state
``` 
starcoin% node network state
```
* Checkout the balance mint with account
```
starcoin% account show
{
  "ok": {
    "account": {
      "address": "0xd9a356781aeb4c87a88f9b6ff7008d65",
      "is_default": true,
      "is_locked": true,
      "is_readonly": false,
      "public_key": "0x0b4613ff6d8fe43c1d5ac65f58b8e7b2c3118d0ee099274af715ce3c098a5733",
      "receipt_identifier": "stc1pmx34v7q6adxg02y0ndhlwqydv5jkf43x"
    },
    "auth_key": "0x44276d8fc703acee45e8f8b00a0e77c2d9a356781aeb4c87a88f9b6ff7008d65",
    "balances": {
      "0x00000000000000000000000000000001::STC::STC": 2130000000000
    },
    "sequence_number": 0
  }
}
```

**Comment by buchii9** [comment]
Are there specific requirements for a user to solo mine FlexiDAG?

**Reply by sanlee42** [reply]
Yes, At least one x86 computer is required, with 1 GB of RAM and a 1-core CPU.

**Comment by buchii9** [comment]
Are We going to be able to use our stbox to mine stc 2.0 ?

**Reply by sanlee42** [reply]
Yes, we can use stc box,the hash algorithm of miner is the same as starcoin 1.0, and it also supports the same stratum protocol.

**Comment by buchii9** [comment]
Where do we paste the codes when we go to IP?

**Reply by sanlee42** [reply]
We need to run the starcoin binary in a terminal

**Comment by buchii9** [comment]
Permission denied. Is something wrong?

![IMG_20231129_170027_440](https://github.com/starcoinorg/starcoin/assets/119014731/9471a812-e321-448a-858e-4e12513c06fc)
](url)

**Reply by sanlee42** [reply]
you can use https protocol to clone,instead of ssh protocol: git clone https://github.com/starcoinorg/starcoin.git

**Comment by sayhisanti** [comment]
Found error in step of connect to seed 
OS: Win10 

2023-12-01T20:31:32.272760500+07:00 INFO - starcoin_rpc_server::service::RpcService service actor started
2023-12-01T20:31:32.283222600+07:00 ERROR - panic occurred:
panic occurred:
2023-12-01T20:31:32.283276900+07:00 ERROR - details: panicked at 'Create cli data dir 
"C:\\Users\\xxx\\.starcoin\\cli\\dag-net:24" fail, err:Os { code: 267, kind: NotADirectory, message: "The 
directory name is invalid." }', cmd\starcoin\src\cli_state.rs:75:37
details: panicked at 'Create cli data dir "C:\\Users\\xxx\\.starcoin\\cli\\dag-net:24" fail, err:Os { code: 
267, kind: NotADirectory, message: "The directory name is invalid." }', cmd\starcoin\src\cli_state.rs:75:37
2023-12-01T20:31:32.283318400+07:00 ERROR - backtrace:    0:     0x7ff757523202 - <unknown>
   1:     0x7ff757523112 - <unknown>

**Reply by sanlee42** [reply]
It's a bug for creating a system directory on windows, we will fix it on next version, please try starcoin on macos or linux currently.

**Comment by buchii9** [comment]
i get wrong output for  ./starcoin_miner --user <user> -n 10

**Reply by sanlee42** [reply]
Try setting miner threads by adding  `--miner-thread 10`  after `starcoin` command

**Reply by buchii9** [reply]
when loading main console?
so for instance, do you mean ./starcoin -n main console --miner-thread 10
Or  ./starcoin_miner --miner-thread 10

**Reply by sanlee42** [reply]
Trying starcoin daemon: ./starcoin -n main --miner-thread 10 console

**Reply by buchii9** [reply]
loads up into console but no mining output, htop indicates cpu not doing much

**Reply by sanlee42** [reply]
The node can start minting after synchronizing all blocks， please check the synchronization status.

**Reply by Jerobech** [reply]
I know yoy gave confirmation about mining with stc box but, do you think with those extra settings needed above. It still works because in the goldshell stc box you cannot set anything accept pool and user and password

**Reply by sanlee42** [reply]
stc box need use stratum protocol for mining. Try setting --stratum-address  and --stratum-port for starcoin node.
